mm2src/lp_ordermatch.rs

```diff
@@ -2531,25 +2551,35 @@ pub async fn lp_ordermatch_loop(ctx: MmArc) {
         }
 
         {
-            let my_maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
+            let mut my_maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
+            let mut to_cancel = vec![];
+
             for (uuid, order) in my_maker_orders.iter() {
                 if !ordermatch_ctx.orderbook.lock().await.order_set.contains_key(uuid) {
-                    if let Ok(Some(_)) = lp_coinfind(&ctx, &order.base).await {
-                        if let Ok(Some(_)) = lp_coinfind(&ctx, &order.rel).await {
-                            let topic = orderbook_topic_from_base_rel(&order.base, &order.rel);
-                            if !ordermatch_ctx.orderbook.lock().await.is_subscribed_to(&topic) {
-                                let request_orderbook = false;
-                                if let Err(e) =
-                                    subscribe_to_orderbook_topic(&ctx, &order.base, &order.rel, request_orderbook).await
-                                {
-                                    log::error!("Error {} on subscribing to orderbook topic {}", e, topic);
-                                }
+                    if let Ok(Some((base, rel))) = find_pair(&ctx, &order.base, &order.rel).await {
+                        if let Err(e) = order.check_balance(&ctx, &base, &rel).await {
+                            log::info!("Error {} on balance check to kickstart order {}, cancelling", e, uuid);
+                            to_cancel.push(*uuid);
+                            continue;
+                        }
+
+                        let topic = orderbook_topic_from_base_rel(&order.base, &order.rel);
+                        if !ordermatch_ctx.orderbook.lock().await.is_subscribed_to(&topic) {
+                            let request_orderbook = false;
+                            if let Err(e) =
+                                subscribe_to_orderbook_topic(&ctx, &order.base, &order.rel, request_orderbook).await
+                            {
+                                log::error!("Error {} on subscribing to orderbook topic {}", e, topic);
                             }
-                            maker_order_created_p2p_notify(ctx.clone(), order).await;
                         }
+                        maker_order_created_p2p_notify(ctx.clone(), order).await;
                     }
                 }
             }
+
+            for uuid in to_cancel {
+                my_maker_orders.remove(&uuid);
```

Comment:
shouldn't we call `maker_order_cancelled_p2p_notify` here?

Reply:
It is done on MM2 restart so it is most likely that other nodes don't have this order in their order books by this time. So we either notify others about order creation or cancel it and do nothing on the P2P side :slightly_smiling_face: